### PR TITLE
adis16497: add sensor model autodetection

### PR DESCRIPTION
**Pull request by mhkabir**
The sensor comes in three variants with different gyro measurement ranges. This patch allows auto-detection of the model at startup and sets the scale factor associated with the particular model.

### CODE REVIEW

**Comment by mhkabir** [comment]
Needs a test on HW.

**Comment by mhkabir** [comment]
@dagar Ready for merge after review.

File: src/drivers/imu/adis16497/ADIS16497.cpp [hunk 1]
```diff
@@ -247,7 +249,19 @@ ADIS16497::probe()
 			PX4_DEBUG("PRODUCT: %X", product_id);
 
 			if (self_test()) {
-				return PX4_OK;
+
+				// Switch to config page
+				write_reg16(PAGE_ID, 0x03);
```
Comment (dagar) [inline src/drivers/imu/adis16497/ADIS16497.cpp]: You don't need to switch back?

File: src/drivers/imu/adis16497/ADIS16497.cpp [hunk 2]
```diff
@@ -53,11 +53,15 @@ static constexpr uint8_t CONFIG  	= 0x0A; // Control, clock and miscellaneous co
...
 static constexpr uint8_t SYNC_SCALE  	= 0x10; // Control, automatic bias correction configuration
-static constexpr uint8_t RANG_MDL   	= 0x12; // Measurement range (model-specific) Identifier TODO use this
+static constexpr uint8_t RANG_MDL   	= 0x12; // Measurement range (model-specific) identifier
 static constexpr uint8_t FILTR_BNK_0   	= 0x16; // Filter selection
 static constexpr uint8_t FILTR_BNK_1   	= 0x18; // Filter selection
 
-static constexpr uint16_t PROD_ID_ADIS16497 = 0x4071;	// ADIS16497 Identification, device number
+static constexpr uint16_t PROD_ID_ADIS16497 = 0x4071; // ADIS16497 device number
+
+static constexpr uint16_t RANG_MDL_1BMLZ = 0x03; // ADIS16497 model number
```
Comment (dagar) [inline src/drivers/imu/adis16497/ADIS16497.cpp]: The comments are generic and repeated. You could make them match the datasheet.
```suggestion
static constexpr uint16_t RANG_MDL_1BMLZ = 0b0011; // ADIS16497-1 (±125°/sec)
```

File: src/drivers/imu/adis16497/ADIS16497.cpp [hunk 3]
```diff
@@ -53,11 +53,15 @@ static constexpr uint8_t CONFIG  	= 0x0A; // Control, clock and miscellaneous co
...
+static constexpr uint16_t RANG_MDL_2BMLZ = 0x07; // ADIS16497 model number
```
Comment (dagar) [inline src/drivers/imu/adis16497/ADIS16497.cpp]:
```suggestion
static constexpr uint16_t RANG_MDL_2BMLZ = 0b0111; // ADIS16497-2 (±450°/sec)
```

File: src/drivers/imu/adis16497/ADIS16497.cpp (hunk 1 above)
Comment (mhkabir) [reply src/drivers/imu/adis16497/ADIS16497.cpp]: Wherever config params are touched, the code always switches to relevant page first. So don't need to explicitly do it here.

File: src/drivers/imu/adis16497/ADIS16497.cpp [hunk 4]
```diff
@@ -247,7 +249,19 @@ ADIS16497::probe()
 			PX4_DEBUG("PRODUCT: %X", product_id);
 
 			if (self_test()) {
-				return PX4_OK;
+
+				// Switch to config page
+				write_reg16(PAGE_ID, 0x03);
+
+				uint16_t model_id = read_reg16(RANG_MDL);
```
Comment (dagar) [inline src/drivers/imu/adis16497/ADIS16497.cpp]: You might want to clear everything but the lower 4 bits. The datasheet just says the reset aren't used.

File: src/drivers/imu/adis16497/ADIS16497.cpp [hunk 5]
```diff
@@ -53,11 +53,15 @@ static constexpr uint8_t CONFIG  	= 0x0A; // Control, clock and miscellaneous co
...
+static constexpr uint8_t RANG_MDL   	= 0x12; // Measurement range (model-specific) identifier
 static constexpr uint8_t FILTR_BNK_0   	= 0x16; // Filter selection
 static constexpr uint8_t FILTR_BNK_1   	= 0x18; // Filter selection
 
-static constexpr uint16_t PROD_ID_ADIS16497 = 0x4071;	// ADIS16497 Identification, device number
+static constexpr uint16_t PROD_ID_ADIS16497 = 0x4071; // ADIS16497 device number
+
+static constexpr uint16_t RANG_MDL_1BMLZ = 0x03; // ADIS16497 model number
+static constexpr uint16_t RANG_MDL_2BMLZ = 0x07; // ADIS16497 model number
+static constexpr uint16_t RANG_MDL_3BMLZ = 0x0F; // ADIS16497 model number
```
Comment (dagar) [inline src/drivers/imu/adis16497/ADIS16497.cpp]:
```suggestion
static constexpr uint16_t RANG_MDL_3BMLZ = 0b1111; // ADIS16497-3 (±2000°/sec)
```

**Pull request merged**